plugin/pkg/admission/allowunsafedelete/admission.go

```diff
@@ -0,0 +1,141 @@
+	if p.authz == nil {
+		return fmt.Errorf("%s requires an authorizer", PluginName)
+	}
+	return nil
+}
+
+// Validate ensures that the user has permission to do 'delete-ignore-read-errors'
+// on the resource being deeted when ignoreStoreReadErrorWithClusterBreakingPotential
+// is enabled, these are the constraints:
+// a) must be a DELETE operation
+// b) ignoreStoreReadErrorWithClusterBreakingPotential is set to true
+// c) the request is a resource, no sub-resource or non-resource endpoint
+// d) the user has permission to do 'delete-ignore-read-errors' on the resource
+func (p *AllowUnsafeDelete) Validate(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	// if the feature is disabled, this plugin, although enabled should not
+	// be active, and requests with the delete option
+	// 'ignoreStoreReadErrorWithClusterBreakingPotential' enabled will
+	// fallback to using the normal deletion flow
+	if !utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AllowUnsafeMalformedObjectDeletion) {
+		return nil
+	}
+	if attr.GetOperation() != admission.Delete || attr.GetOperationOptions() == nil {
+		return nil
+	}
+
+	options, ok := attr.GetOperationOptions().(*metav1.DeleteOptions)
+	if !ok {
+		return fmt.Errorf("expected an option of type: %T, but got: %T", &metav1.DeleteOptions{}, attr.GetOperationOptions())
+	}
+	if !ptr.Deref(options.IgnoreStoreReadErrorWithClusterBreakingPotential, false) {
+		return nil
+	}
+
+	requestInfo, found := request.RequestInfoFrom(ctx)
+	if !found {
+		return admission.NewForbidden(attr, errors.New("no RequestInfo found in the context"))
+	}
+	if !requestInfo.IsResourceRequest || len(attr.GetSubresource()) > 0 {
+		return admission.NewForbidden(attr, errors.New("ignoreStoreReadErrorWithClusterBreakingPotential delete option is not allowed on a subresource or non-resource request"))
+	}
+
+	// if we are here, IgnoreStoreReadErrorWithClusterBreakingPotential
+	// is set to true in the delete options, the user must have permission
+	// to do 'delete-ignore-read-errors' on the given resource.
+	const verb = "delete-ignore-read-errors"
+	record := authorizer.AttributesRecord{
+		User:            attr.GetUserInfo(),
+		Verb:            verb,
+		Namespace:       attr.GetNamespace(),
+		Name:            attr.GetName(),
+		APIGroup:        attr.GetResource().Group,
+		APIVersion:      attr.GetResource().Version,
+		Resource:        attr.GetResource().Resource,
+		ResourceRequest: true,
+	}
+
+	decision, reason, err := p.authz.Authorize(ctx, record)
+	if err != nil {
+		err = fmt.Errorf("error while checking permission for %q, %w", verb, err)
+		klog.FromContext(ctx).V(1).Error(err, "failed to authorize")
+		return admission.NewForbidden(attr, err)
+	}
+	if decision == authorizer.DecisionAllow {
+		// TODO: no distinguishable attribute in the audit entry to
+		// discern whether it was a normal deletion flow or unsafe,
+		// add an annotation to audit to make that distinction?
```

Comment:
Yes.  use the same annotation as the storage I'd think.

Reply:
so for the admission, i have the following annotations that follow the admission convention:
```
allowunsafemalformedobjectdeletion.admission.k8s.io/decision: "allow|deny|empty when erroroccurs"
allowunsafemalformedobjectdeletion.admission.k8s.io/reason: "reason from the authorizer|internal error"
```

for the storage, I am using:
```
apiserver.k8s.io/unsafe-delete-ignore-read-error: ""
```

Let me know if you want to change it